src/modules/pages/components/RouteLayouts/UserNavigation.tsx

```diff
@@ -115,8 +145,177 @@ const UserNavigation = () => {
+
+  const networkNameForReputationOracle =
+    process.env.NODE_ENV === 'development'
+      ? 'local'
+      : NETWORK_DATA[
+          process.env.NETWORK || DEFAULT_NETWORK
+        ].shortName.toLowerCase(); // @TODO have to make sure this will work for other networks than xdai
+
+  useEffect(() => {
+    let latestBlockFromRpc: string;
+
+    const networkCheck = setInterval(async () => {
+      const rpcBlockResponsePromise = raceAgainstTimeout(
+        fetch(rpcEndpoint, {
+          method: 'POST',
+          body: JSON.stringify({
+            jsonrpc: '2.0',
+            method: 'eth_getBlockByNumber',
+            params: ['latest', false],
+            id: 1,
+          }),
+        }),
+        TIMEOUT,
+        new Error('Timeout reached trying to get latest block from RPC'),
+      );
+
+      // Is the server alive?
+      const serverResponsePromise = raceAgainstTimeout(
+        fetch(`${serverEndpoint}/liveness`),
+        TIMEOUT,
+        new Error('Timeout reached trying to reach server'),
+      );
+
+      const reputationOracleResponsePromise = raceAgainstTimeout(
+        fetch(`${reputationOracleEndpoint}/${networkNameForReputationOracle}`),
+        TIMEOUT,
+        new Error('Timeout reached trying to reach reputation oracle.'),
+      );
+
+      const latestBlockFromSubgraphPromise = raceAgainstTimeout(
+        getLatestSubgraphBlock(apolloClient),
+        TIMEOUT,
+        new Error('Timeout reached trying to get latest block from subgraph'),
+      );
+
+      try {
+        const [
+          rpcBlockResponse,
+          latestBlockFromSubgraph,
+          serverResponse,
+          reputationOracleResponse,
+        ] = await Promise.all([
+          rpcBlockResponsePromise,
+          latestBlockFromSubgraphPromise,
+          serverResponsePromise,
+          reputationOracleResponsePromise,
+        ]);
+
+        if (rpcBlockResponse.status !== 200) {
+          setNetworkHealth('critical');
+          return;
+        }
+        latestBlockFromRpc = await rpcBlockResponse
+          .json()
+          .then((r) => r.result.number); // this is in hex
+
+        // ? @TODO Reputation Oracle status is not enough - also check if it is behind
+
+        // @TODO Is IPFS endpoints alive?
+
+        if (
+          parseInt(latestBlockFromRpc, 16) > latestBlockFromSubgraph ||
+          serverResponse.status !== 200 ||
+          reputationOracleResponse.status !== 200
+        ) {
+          setNetworkHealth('poor');
+        } else {
+          // If everything is okay, set health to healthy (to correct for previous state)
+          setNetworkHealth('healthy');
+        }
+      } catch (err) {
+        setNetworkHealth('poor');
+      }
+    }, networkCheckInterval);
+    return () => clearInterval(networkCheck);
+  }, []);
+
   return (
     <div className={styles.main}>
+      {networkHealth !== 'healthy' && (
+        <div>
+          <MiniSpinnerLoaderWrapper milliseconds={networkHealthLoadingTime}>
```

Comment:
Why is this decoupled in terms of logic?

Why not hide it when everything is set up properly here ?